app/Models/Post.php

```diff
@@ -361,32 +352,32 @@ public function toCustomerIoPayload()
             $school = app(GraphQL::class)->getSchoolById($this->school_id);
         }
 
-        // The associated Action for this post.
         $action = $this->actionModel;
+        $signup = optional($this->signup);
```

Comment:
This fixes the problems we'd run into with this method & posts without signups, since [`optional`](https://laravel.com/docs/8.x/helpers#method-optional) always returns an object that you can work with (and if the signup doesn't exist, properties or method calls will simply return `null`).